app/forms/hub/update_client_form.rb

```diff
@@ -39,13 +41,23 @@ class UpdateClientForm < ClientForm
     validates :state_of_residence, inclusion: { in: States.keys }
     validates :preferred_interview_language, presence: true, allow_blank: false
 
+    validates :primary_tin_type, presence: true
+    validates :primary_ssn, social_security_number: true, if: -> { ["ssn", "ssn_no_employment"].include? primary_tin_type }
+    validates :primary_ssn, individual_taxpayer_identification_number: true, if: -> { primary_tin_type == "itin"}
+
+    validates_confirmation_of :spouse_ssn, if: -> { filing_joint == "yes" }
+    validates :spouse_ssn, social_security_number: true, if: -> { ["ssn", "ssn_no_employment"].include?(spouse_tin_type) && filing_joint == "yes"}
+    validates :spouse_ssn, individual_taxpayer_identification_number: true, if: -> { spouse_tin_type == "itin" && filing_joint == "yes" }
+
+    attr_accessor :client
```

Comment:
Dumb question: What's the value of adding this?

Reply:
in the "ctc only" part of the view (that form already includes this line), we call @form.client to get some logic around whether the confirmation field for ssn should exist. Now that we're reusing that logic, we need to add this.